### PR TITLE
feat: scale and map body

### PR DESCRIPTION
**Pull request by jonahrb**
## Description
Ability to scale or map a body

## Issue linked
#1036 

## Checklist
- [x] I have tested my changes locally.
- [x] I have added necessary documentation or updated existing documentation.
- [x] I have followed the coding style guidelines of this project.
- [x] I have added appropriate unit tests.
- [x] I have reviewed my changes before submitting this pull request.
- [x] I have linked the issue or issues that are solved to the PR if any.
- [x] I have assigned this PR to myself.
- [x] I have added the minimum version decorator to any new backend method implemented.
- [x] I have made sure that the title of my PR follows [Conventional commits style](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. ``feat: extrude circle to cylinder``)

### CODE REVIEW

File: src/ansys/geometry/core/designer/body.py [hunk 1]
```diff
@@ -358,6 +362,42 @@ def rotate(
         """
         return
 
+    @abstractmethod
+    def scale(
+        self,
+        value: Real,
+    ) -> None:
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:369]: I'd single line them if possible
```suggestion
    def scale(self, value: Real) -> None:
```

File: src/ansys/geometry/core/designer/body.py [hunk 2]
```diff
@@ -358,6 +362,42 @@ def rotate(
...
+        """
+        Scale the geometry body by the given value.
+
+        Parameters
+        ----------
+        value: Real
+            Value to scale the body by.
+        Returns
+        -------
+        None
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:379]: If nothing is returned, no need to add the return section

```suggestion
            Value to scale the body by.
```

File: src/ansys/geometry/core/designer/body.py [hunk 3]
```diff
@@ -358,6 +362,42 @@ def rotate(
...
+        """
+        return
+
+    @abstractmethod
+    def mirror(
+        self,
+        frame: Frame,
+    ) -> None:
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:387]:
```suggestion
    @abstractmethod
    def mirror(self, frame: Frame) -> None:
```

File: src/ansys/geometry/core/designer/body.py [hunk 4]
```diff
@@ -358,6 +362,42 @@ def rotate(
...
+        """
+        Mirror the geometry body across the specified frame.
+
+        Parameters
+        ----------
+        frame: Frame
+            Structure defining the mirroring/mapping of the body.
+        Returns
+        -------
+        None
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:397]:
```suggestion
            Structure defining the mirroring/mapping of the body.
```

File: src/ansys/geometry/core/designer/body.py [hunk 5]
```diff
@@ -818,6 +858,25 @@ def rotate(
             )
         )
 
+    @protect_grpc
+    @check_input_types
+    @reset_tessellation_cache
+    @min_backend_version(24, 1, 0)
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:864]: Oh! It was already available for all backends?

File: src/ansys/geometry/core/designer/body.py [hunk 6]
```diff
@@ -818,6 +858,25 @@ def rotate(
...
+        self._bodies_stub.Scale(ScaleRequest(id=self.id, scale=value))
+
+    @protect_grpc
+    @check_input_types
+    @reset_tessellation_cache
+    @min_backend_version(24, 1, 0)
+    def mirror(
+        self,
+        frame: Frame,
+    ) -> None:  # noqa: D102
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:876]:
```suggestion
    def mirror(self, frame: Frame) -> None:  # noqa: D102
```

File: src/ansys/geometry/core/designer/body.py (hunk 5 above)
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:864]: If that is the case (it was already available in 24R1) no need to add the decorator. We assume the first version is 24R1.
```suggestion
```

File: src/ansys/geometry/core/designer/body.py [hunk 7]
```diff
@@ -818,6 +858,25 @@ def rotate(
...
+    @reset_tessellation_cache
+    @min_backend_version(24, 1, 0)
+    def scale(self, value: Real) -> None:  # noqa: D102
+        self._grpc_client.log.debug(f"Scaling body {self.id}.")
+        self._bodies_stub.Scale(ScaleRequest(id=self.id, scale=value))
+
+    @protect_grpc
+    @check_input_types
+    @reset_tessellation_cache
+    @min_backend_version(24, 1, 0)
```
Comment (RobPasMue) [inline src/ansys/geometry/core/designer/body.py:872]: Same as before here
```suggestion
```

**Review by RobPasMue — APPROVED** [review]
Looking good @jonahrb

File: src/ansys/geometry/core/designer/body.py (hunk 5 above)
Comment (jonahrb) [reply src/ansys/geometry/core/designer/body.py]: @RobPasMue No, but i was able to retroactively add it, so it is technically available with 24.1 but they would need an updated service :)

Comment (RobPasMue) [reply src/ansys/geometry/core/designer/body.py]: In that case... let's avoid "retroactive" behavior. I talked about this with @b-matteo. Your service is the one that defines the backend version number so you need at least a 24R2 service to do this. Meaning that we need to add the decorators as 24, 2, 0

Comment (jonahrb) [reply src/ansys/geometry/core/designer/body.py]: @RobPasMue makes sense, on it

**Comment by jonahrb** [comment]
@RobPasMue FYI i renamed "mirror" to "map" since that's a more accurate description of the method. I also added unit tests, see `test_body_mapping()` for a couple examples

File: tests/integration/test_design.py [hunk 8]
```diff
@@ -1927,3 +1927,111 @@ def test_get_collision(modeler: Modeler):
 
     assert body1.get_collision(body2) == CollisionType.TOUCH
     assert body2.get_collision(body3) == CollisionType.NONE
+
+
+def test_body_scale(modeler: Modeler):
+    """Test for verifying the correct scaling of a ``Body``."""
```
Comment (PipKat) [inline tests/integration/test_design.py:1933]:
```suggestion
    """Verify the correct scaling of a body."""
```

File: tests/integration/test_design.py [hunk 9]
```diff
@@ -1927,3 +1927,111 @@ def test_get_collision(modeler: Modeler):
...
+
+    body.scale(2)
+    assert Accuracy.length_is_equal(body.volume.m, 8)
+
+    body.scale(0.25)
+    assert Accuracy.length_is_equal(body.volume.m, 1 / 8)
+
+
+def test_body_mapping(modeler: Modeler):
+    """Test for verifying the correct mapping of a ``Body``."""
```
Comment (PipKat) [inline tests/integration/test_design.py:1948]:
```suggestion
    """Verify the correct mapping of a body."""
```

File: tests/integration/test_design.py (hunk 8 above)
Comment (RobPasMue) [reply tests/integration/test_design.py:1933]: a body works @PipKat - thanks for pointing it out nonetheless :)

**Review by RobPasMue — DISMISSED** [review]
Tests are great, implementation is great. Thanks a lot @jonahrb - this is going to be extremely helpful. I would only suggest adding some examples in our documentation on how to do these operations. But it can be done in another PR. I will create an issue and assign it to you.

**Review by dastan-ansys — DISMISSED** [review]
looks good! I am not big on documentation personally, but I feel like it would be a good idea to add a comment that describes the concrete implementation of scale() and map() or links them back the comment on their abstract counterparts.

**Review by b-matteo — APPROVED** [review]

**Pull request merged**